src/playlist-maker.js

```diff
@@ -205,10 +267,23 @@ export default function factory(player, initialList, initialIndex = 0) {
     }
   });
 
+  player.on('duringplaylistchange', () => {
+    if (Array.isArray(list)) {
+      generatePlaylistItemId(list);
+    }
+  });
+
+  player.on('playlistitem', (event, hashopt) => {
```

Comment:
best to just call it `playlistItem` or `item` instead of `hashopt`, less confusing.